functions/Export-SqlUser.ps1

```diff
@@ -250,50 +251,156 @@ https://dbatools.io/Export-SqlUser
                                 if ($DatabasePermission.PermissionState -eq "GrantWithGrant")
                                 {
                                     $WithGrant = "WITH GRANT OPTION"
+                                    $GrantDatabasePermission = 'GRANT'
                                 } 
                                 else 
                                 {
                                     $WithGrant = ""
+                                    $GrantDatabasePermission = $DatabasePermission.PermissionState.ToString().ToUpper()
                                 }
-                                $GrantDatabasePermission = $DatabasePermission.PermissionState.ToString().Replace("WithGrant", "").ToUpper()
 
-                                $outsql += "$($GrantDatabasePermission) $($DatabasePermission.PermissionType) TO [$($DatabasePermission.Grantee)] $WithGrant"
+                                $outsql += "$($GrantDatabasePermission) $($DatabasePermission.PermissionType) TO [$($DatabasePermission.Grantee)]$($WithGrant) AS [$($DatabasePermission.Grantor)];"
                             }
 
 
 	                        #Database Object Permissions
-                            foreach ($ObjectPermission in $db.EnumObjectPermissions() | Where-Object {@("sa","dbo","information_schema","sys") -notcontains $_.Grantee -and $_.Grantee -notlike "##*" -and $dbuser.Name -contains $_.Grantee})
+                            # NB: This is a bit of a mess for a couple of reasons
+                            # 1. $db.EnumObjectPermissions() doesn't enumerate all object types
```

Comment:
Yes, thank you! The other commands that use this will need to be updated as well.